compiler-plugin/src/main/java/io/ballerina/wso2/apim/catalog/codemodifier/OpenAPIAnnotationModifier.java

```diff
@@ -0,0 +1,258 @@
+        for (Module module: currentPackage.modules()) {
+            Collection<DocumentId> documentIds = module.documentIds();
+            for (DocumentId documentId: documentIds) {
+                modifyDocument(context, currentPackage, module, documentIds, documentId);
+            }
+        }
+    }
+
+    private void modifyDocument(SourceModifierContext context, Package currentPackage, Module module,
+                                Collection<DocumentId> documentIds, DocumentId documentId) {
+        Document document = module.document(documentId);
+        SyntaxTree syntaxTree = document.syntaxTree();
+        ModulePartNode rootNode = syntaxTree.rootNode();
+        SemanticModel semanticModel = context.compilation().getSemanticModel(module.moduleId());
+        NodeList<ImportDeclarationNode> newImports = updateImports(rootNode.imports(), rootNode.members());
+
+        NodeList<ModuleMemberDeclarationNode> newMembers = updateMemberNodes(
+                syntaxTree, currentPackage, document,
+                currentPackage.project(), rootNode.members(), semanticModel);
+        ModulePartNode newModulePart = rootNode.modify(newImports, newMembers, rootNode.eofToken());
+        SyntaxTree updatedSyntaxTree = syntaxTree.modifyWith(newModulePart);
+        TextDocument textDocument = updatedSyntaxTree.textDocument();
+        if (documentIds.contains(documentId)) {
+            context.modifySourceFile(textDocument, documentId);
+        } else {
+            context.modifyTestSourceFile(textDocument, documentId);
+        }
+    }
+
+    private NodeList<ImportDeclarationNode> updateImports(NodeList<ImportDeclarationNode> imports,
+                                                          NodeList<ModuleMemberDeclarationNode> members) {
+        boolean containsServiceNode = false;
+        for (ModuleMemberDeclarationNode memberNode : members) {
+            if (memberNode.kind() != SyntaxKind.SERVICE_DECLARATION) {
+                continue;
+            }
+            containsServiceNode = true;
+            ServiceDeclarationNode serviceNode = (ServiceDeclarationNode) memberNode;
+            NodeList<AnnotationNode> annotations = getMetadataNode(serviceNode).annotations();
+            for (AnnotationNode annotation: annotations) {
+                if (isServiceCatalogConfigAnnotationAvailable(annotation)) {
+                    return imports;
+                }
+            }
+        }
+        if (containsServiceNode) {
+            return imports.add(createImportDeclarationNodeForModule());
+        }
+        return imports;
+    }
+
+    private NodeList<ModuleMemberDeclarationNode> updateMemberNodes(
+            SyntaxTree syntaxTree, Package currentPackage, Document document, Project project,
+            NodeList<ModuleMemberDeclarationNode> oldMembers, SemanticModel semanticModel) {
+        List<ModuleMemberDeclarationNode> updatedMembers = new LinkedList<>();
```

Comment:
Is there a specific reason for using a `LinkedList` here. Given `createNodeList` accepts an `Collection` I assume `ArrayList` should be okay?. If there is a specific reason lets add a comment here.